wordninja.py

```diff
@@ -25,11 +25,23 @@
 #   <list of contractions>
 
 
+class FileTypeMagicBytesRe():
+  BZIP_FILE = re.compile(b'^\\x42\\x5a\\x68')
+  GZIP_FILE = re.compile(b'^\\x1f\\x8b\\x08')
+
+
 class LanguageModel(object):
   def __init__(self, word_file):
     # Build a cost dictionary, assuming Zipf's law and cost = -math.log(probability).
-    with gzip.open(word_file) as f:
-      words = f.read().decode().split()
+    if check_magic(word_file, FileTypeMagicBytesRe.BZIP_FILE):
```

Comment:
since we have the filename, why not just check the extension?

Reply:
Just in case there's any discrepancy between the file extension and its type. Although that really depends on what's the expected behavior if it's fed bad inputs.

Should it:
1. Accept the extension and just throw out errors if it fails to process?
2. Ignore the extension and just process it best it can?
3. Do #2, but pop out a warning if the extension doesn't match?